frontend/lib/src/dataframes/arrowConcatUtils.ts

```diff
@@ -28,10 +28,38 @@ import {
   PandasColumnType,
   PandasIndexTypeName,
   PandasRangeIndex,
-  sameDataTypes,
-  sameIndexTypes,
 } from "./arrowTypeUtils"
 
+/** True if both arrays contain the same data types in the same order. */
+function sameDataTypes(
+  t1: PandasColumnType[],
+  t2: PandasColumnType[]
+): boolean {
+  // NOTE: We remove extra columns from the DataFrame that we add rows from.
+  // Thus, as long as the length of `t2` is >= than `t1`, this will work properly.
+  // For columns, `pandas_type` will point us to the correct type.
+  return t1.every(
+    (type: PandasColumnType, index: number) =>
+      type.pandas_type === t2[index]?.pandas_type
+  )
+}
+
+/** True if both arrays contain the same index types in the same order. */
```

Comment:
In contrast to `sameDataTypes`, I would state in the JSDoc that `If the arrays have different lengths, they are never the same`

Reply:
Added 👍